csharp/test/TestCommon/TestHelper.cs

```diff
@@ -284,3 +284,260 @@ public static async Task<int> RunTestAsync<T>(string[] args) where T : TestHelpe
         }
     }
 }
+
+// The new version of the Test helper classes. Will be renamed Test once all the tests have migrated to TestNew.
+namespace TestNew
+{
+    public abstract class TestHelper
+    {
+        private TextWriter? _writer;
+
+        public ushort BasePort => GetTestBasePort(Communicator.GetProperties());
+
+        public Communicator Communicator { get; private init; } = null!;
+
+        public ZeroC.Ice.Encoding Encoding => GetTestEncoding(Communicator.GetProperties());
+
+        public string Host => GetTestHost(Communicator.GetProperties());
+
+        public Protocol Protocol => GetTestProtocol(Communicator.GetProperties());
+
+        public string Transport => GetTestTransport(Communicator.GetProperties());
+
+        public TextWriter Output
+        {
+            get => _writer ?? Console.Out;
+            set => _writer = value;
+        }
+
+        public static void Assert([DoesNotReturnIf(false)] bool b, string message = "")
+        {
+            if (!b)
+            {
+                Debug.Assert(false, message);
+                throw new Exception(message);
+            }
+        }
+
+        public static Communicator CreateCommunicator(
+            ref string[] args,
+            Dictionary<string, string>? defaults = null,
+            ZeroC.Ice.Instrumentation.ICommunicatorObserver? observer = null) =>
+            CreateCommunicator(CreateTestProperties(ref args, defaults), observer);
+
+        public static Communicator CreateCommunicator(
+            Dictionary<string, string> properties,
+            ZeroC.Ice.Instrumentation.ICommunicatorObserver? observer = null)
+        {
+            var tlsServerOptions = new TlsServerOptions();
+            if (properties.TryGetValue("Test.Transport", out string? value))
+            {
+                // When running test with WSS, disable client authentication for browser compatibility
+                tlsServerOptions.RequireClientCertificate = value == "ssl";
+            }
+
+            return new Communicator(properties, tlsServerOptions: tlsServerOptions, observer: observer);
+        }
+
+        public string GetTestEndpoint(int num = 0, string transport = "", bool ephemeral = false) =>
+            GetTestEndpoint(Communicator.GetProperties(), num, transport, ephemeral);
+
+        public static string GetTestEndpoint(
+            Dictionary<string, string> properties, int num = 0, string transport = "", bool ephemeral = false)
```

Comment:
This uses the old style, should be one param per line

Reply:
Fixed.